cmd/beyondctl/utils.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import "strings"
+
+// ParseKeyFromConnectionString parse key and keep connection string as original without object key
+// since we do not check connection string valid or not here,
+// so we will only split key from original connection string
+// and return the key-split connection string, key, not return an error
+func ParseKeyFromConnectionString(input string) (conn, key string) {
+	var b strings.Builder
+
+	question := strings.Index(input, "?")
+	var connStr, paramStr string
+
+	// no question mark found, like: fs:///path/to/dir/key, means no param found
+	if question == -1 {
+		connStr = input
+	} else {
+		connStr = input[:question]
+		paramStr = input[question:] // paramStr keeps ? as its start so that we can build result directly
+	}
+
+	// if connStr end with /, we handle it as a dir, split key from connStr
+	// otherwise, we handle it as /dir/to/key
+	if !strings.HasSuffix(connStr, "/") {
+		split := strings.LastIndex(connStr, "/")
+		key = connStr[split+1:]
+		connStr = connStr[:split+1]
+	}
```

Comment:
I'm not sure whether it is the right approach to split `work_dir` and `key` by `/`. We can't handle keys like `/abc`?

Reply:
We can specify different actions between `/abc` and `/abc/`, users can decide what they want. Otherwise, we cannot get object key from an arg string.

Reply:
How about use other ways? Maybe we can add extra key-value pair or field to carry the key?

Reply:
You mean like `fs:///path/to/work_dir?object_key=file.txt`?

This looks a little bit weird.

Reply:
I prefer introduce #6

Reply:
After #6 implemented, I'll close this PR and implement new one with profile instead.

Reply:
So now, just leave this one here, and record our opinions.